cypress/e2e/alerts.cy.js

```diff
@@ -1,25 +1,60 @@
-describe('Cypress application', () => {
-  before(() => {
+/// <reference types='cypress' />
+const faker = require('faker');
 
+describe('Cypress application', () => {
+  beforeEach(() => {
+    cy.visit('/');
   });
 
-  it('should have the ability to assert automatically resolved alerts', () => {
+  it('should be able to assert alerts that are resolved automatically', () => {
+    cy.get('#alertButton').click();
 
+    cy.on('window:alert', (alertText) => {
+      expect(alertText).to.equal('You clicked a button');
+    });
   });
 
-  it('should have the ability to assert scheduled allert', () => {
+  it('should be able to assert alerts that are scheduled', () => {
+    cy.get('#timerAlertButton').click();
+    cy.clock();
+    cy.tick(5000);
 
+    cy.on('window:alert', (alertText) => {
+      expect(alertText).to.equal('This alert appeared after 5 seconds');
+    });
   });
 
-  it('should autimatically resolve alerts', () => {
-
+  it('should randomly resolve or cancel alerts', () => {
+    const randomChoice = Math.random() < 0.5;
+
+    cy.get('#confirmButton').click();
+
+    cy.on('window:confirm', (confirmText) => {
+      expect(confirmText).to.equal('Do you confirm action?');
+      return randomChoice;
+    });
+
+    cy.wrap(randomChoice).then((choice) => {
+      if (choice) {
+        cy.contains('#confirmResult', 'You selected Ok').then(() => {
+          cy.log('User chose: Ok');
+        });
+      } else {
+        cy.contains('#confirmResult', 'You selected Cancel').then(() => {
+          cy.log('User chose: Cancel');
+        });
+      }
+    });
   });
```

Comment:
This test case is a bit confusing. It's not clear what it means for alerts to be 'randomly resolved or cancelled'. It would be helpful to provide a more detailed description.